### DISCUSSION TITLE
Align IDIDManager interface with DID Registration spec

### DISCUSSION THREAD

**Discussion by peacekeeper** [decentralized-identity/veramo#959] [Ideas]
Hello,

At DIF we've been working on a specification called "DID Registration", which defines inputs and outputs of the DID create/update/deactivate operations, see here:
https://identity.foundation/did-registration/

Just like DID Resolution, these are abstract functions which can be implemented in various ways, e.g. as a local SDK, remote HTTP service, etc. There is also the Universal Registrar implementation of this at DIF: https://github.com/decentralized-identity/universal-registrar

The DID Registration specification seems to have some overlap with Veramo's [IDIDManager](https://veramo.io/docs/api/core.ididmanager/) interface. Perhaps it would be useful to align your interface with the specification, or conversely, incorporate ideas from your interface into the specification.

Markus

**Comment by mirceanis** [comment]
Thank you for bringing this to our attention, I think it makes a lot of sense to try to implement this, or at least to try to align the concepts.

I'll repost this as an issue since it's very likely we'll start prioritizing it: #960
